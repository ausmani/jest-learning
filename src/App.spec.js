import React from 'react';
import renderer from 'react-test-renderer';
import App, {Counter, dataReducer} from "./App";

const list = ['a', 'b', 'c'];


describe('App', ()=>{
    // test('Snapshot app render', ()=>{
    //     const component = renderer.create(<App/>);
    //     let tree = component.toJSON();
    //     expect(tree).toMatchSnapshot();
    // });
    describe('Reducer',()=>{
        it('should set a list', function () {
            const state ={list:[],error:null};
            const newState = dataReducer(state,{type:'SET_LIST',list});
            expect(newState).toEqual({list,error:null})
        });
        it('should reset the error if list is set', () => {
            const state = { list: [], error: true };
            const newState = dataReducer(state, {
                type: 'SET_LIST',
                list,
            });

            expect(newState).toEqual({ list, error: null });
        });

        it('should set the error', () => {
            const state = { list: [], error: null };
            const newState = dataReducer(state, {
                type: 'SET_ERROR',
            });

            expect(newState.error).toBeTruthy();
        });

    })
})
describe('Checking Counter Component',()=>{

    test('Snapshot Renderers',()=>{
        const component = renderer.create(<Counter counter={1}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})



