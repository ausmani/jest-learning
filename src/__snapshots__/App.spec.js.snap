// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking Counter Component Snapshot Renderers 1`] = `
<div>
  <p>
    1
  </p>
</div>
`;
